plugins/bananaslug/src/content_scripts/ScriptInjector.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2013 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The ScriptInjector file.
+ */
+
+var Fetcher = require('./Fetcher');
```

Comment:
We shouldn't be using XHR requests on pages that we inspect. This creates origin problem, potential security issues and screws up debugging of the page. (E.g. network requests popping up in the network tab.)